### PR TITLE
removing the bundle execution to runtime.

### LINKED ISSUE #4: Move ExecuteApb bits from bundle to runtime
`executor.go` has `ExecuteApb,` this method should move to `runtime` package. But `executor.go` remains in `bundle` package.

### PR DESCRIPTION
**Pull request by shawn-hurley**
fixes #4 
* adding multiple layers of execution now.
* each executor method now controls create sandbox
* executor.executeBundle now does some stuff that is common
* runtime.RunBundle and Copy Secrets are overridable functions for the
runtime.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
Waiting for #60 to be merged before this should be merged

File: bundle/bind.go [hunk 1]
```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
 
 	go func() {
 		e.actionStarted()
-		executionContext, err := e.executeApb(
-			"bind", instance.Spec, instance.Context, parameters)
+		// Create namespace name that will be used to generate a name.
+		ns := fmt.Sprintf(fmt.Sprintf("%s-%.4s-", instance.Spec.FQName, bindAction))
```
Comment (dymurray) [inline bundle/bind.go]: I'm sure there's a reason for doing the double `fmt.Sprintf` but would like to hear what it is :)

File: bundle/bind.go [hunk 2]
```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
...
+			ec.BundleName,
+			ec.Location,
+			ec.Targets,
 			clusterConfig.Namespace,
 			clusterConfig.KeepNamespace,
 			clusterConfig.KeepNamespaceOnError,
 		)
 		if err != nil {
-			log.Errorf("Problem executing apb [%s] bind", executionContext.PodName)
+			log.Errorf("Problem executing apb [%s] bind", ec.BundleName)
```
Comment (dymurray) [inline bundle/bind.go]: `Problem executing bundle [%s] bind` for consistency

**Review by dymurray — COMMENTED** [review]
Same comments as other PR. Other than the nits looks good.

File: bundle/executor.go [hunk 3]
```diff
@@ -156,140 +149,88 @@ func (e *executor) updateDescription(newDescription string, dashboardURL string)
...
-					Env:             e.createPodEnv(executionContext),
-					ImagePullPolicy: pullPolicy,
-					VolumeMounts:    volumeMounts,
-				},
-			},
-			RestartPolicy:      v1.RestartPolicyNever,
-			ServiceAccountName: executionContext.ServiceAccount,
-			Volumes:            volumes,
-		},
+	exContext, err = runtime.Provider.RunBundle(exContext)
```
Comment (dymurray) [inline bundle/executor.go:201]: +1 this is great.

**Review by dymurray — APPROVED** [review]

**Review by jmrodri — APPROVED** [review]
LGTM

**Pull request merged**